modin/core/storage_formats/pandas/utils.py

```diff
@@ -92,23 +92,16 @@ def split_result_of_axis_func_pandas(axis, num_splits, result, length_list=None)
 
     if length_list is not None:
         length_list.insert(0, 0)
-        sums = np.cumsum(length_list)
-        if axis == 0 or isinstance(result, pandas.Series):
-            return [result.iloc[sums[i] : sums[i + 1]] for i in range(len(sums) - 1)]
-        else:
-            return [result.iloc[:, sums[i] : sums[i + 1]] for i in range(len(sums) - 1)]
-
+    else:
+        chunksize = compute_chunksize(result.shape[axis], num_splits)
+        length_list = np.full(num_splits + 1, chunksize)
```

Comment:
This brings an additional overhead, but in the region of 1-2 milliseconds for a `NUM_SPLITS==1000`, I consider it acceptable to improve readability.